vortex-array/src/macros.rs

```diff
@@ -41,27 +41,32 @@ macro_rules! impl_encoding {
                 type Encoding = [<$Name Encoding>];
             }
 
-            #[derive(std::fmt::Debug, Clone)]
-            pub struct [<$Name Array>] {
-                data: $crate::ArrayData,
-                metadata: [<$Name Metadata>],
+            impl $crate::encoding::Encoding for [<$Name Encoding>] {
+                type Array = [<$Name Array>];
+                type Metadata = [<$Name Metadata>];
             }
 
+            #[derive(std::fmt::Debug, Clone)]
+            #[repr(transparent)]
+            pub struct [<$Name Array>]($crate::ArrayData);
```

Comment:
I find it aesthetically pleasing that `<$Name Array>` is just a newtype wrapper for ArrayData now

Reply:
It's a shame, `Array<Encoding>` would mean we could auto-impl almost everything. But it means other crates can't add their own functions to the array.